atum/src/main/scala/za/co/absa/atum/core/ControlFrameworkState.scala

```diff
@@ -252,7 +253,15 @@ class ControlFrameworkState(sparkSession: SparkSession) {
   }
 
   private[atum] def storeCurrentInfoFile(outputInfoFilePath: Path)(implicit outputFs: FileSystem): Unit = {
-    val outputFilePath = if (outputFs.isDirectory(outputInfoFilePath)) {
+    var isDirectory = false
+    try {
+      isDirectory = outputFs.getFileStatus(outputInfoFilePath).isDirectory
+    }
+    catch {
+      case e: FileNotFoundException =>
+        isDirectory = false
+    }
```

Comment:
In _Scala_ most if not all language constructs are expression that can be assigned.

```suggestion
    val isDirectory: Boolean = try {
      outputFs.getFileStatus(outputInfoFilePath).isDirectory
    } catch {
      case _: FileNotFoundException => false
    }
```

Or you can use the `Try` class:
```suggestion
    val isDirectory= Try(outputFs.getFileStatus(outputInfoFilePath).isDirectory)
      .recover{case _: FileNotFoundException => false }.get
```
For this later option the `Try` class have to be imported:
`import scala.util.Try`

Reply:
In the second example:
* Line 256 result is [`Try[Boolean]`](https://danielwestheide.com/blog/the-neophytes-guide-to-scala-part-6-error-handling-with-try/#the-semantics-of-try) type
* Next line we trying to `recover`, in case the `Failure` is of type `FileNotFoundException` we say it's `false` (`Success(false)` actually)
* The `_` is there, because we don't care about the actual exception (like in case of `case e: FileNotFoundException`)
* `.get` at the end on `Try` returns the value held withing `Success` - type `Boolean` in this case. Or throws the exception held in the `Failure`

Also calling this out to attention of @AB006J9

Reply:
Thanks for the detailed explanation. Updated.